app/forms.py

```diff
@@ -0,0 +1,143 @@
+# import re
+from django import forms
+# from PIL import Image
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+# from django.core.exceptions import ValidationError
+import time
+
+
+class RegistrationForm(forms.Form):
+
+    username = forms.RegexField(
+        regex=r'^\w+$',
+        widget=forms.TextInput(attrs=dict(required=True, max_length=30)),
+        label=_("Username"),
+        error_messages={'invalid': _("This value must contain only letters, \
+         numbers and underscores.")})
+    password1 = forms.CharField(
+        widget=forms.PasswordInput(
+            attrs=dict(
+                required=True,
+                max_length=4,
+                render_value=False,
+                type='password')),
+        label=_("PIN"))
+    year = forms.IntegerField(
+        widget=forms.TextInput(attrs={'type': 'number'}),
+        label=_("Year"))
+    month = forms.IntegerField(
+        widget=forms.TextInput(attrs={'type': 'number'}),
+        label=_("Month"))
+    day = forms.IntegerField(
+        widget=forms.TextInput(attrs={'type': 'number'}),
+        label=_("Day"))
```

Comment:
Would it make sense to make these ChoiceFields with sensible defaults? Would again save on validation and we can rely on the phone's own widget set which could arguably be a better UX.